guides/keras_cv/classification_with_keras_cv.py

```diff
@@ -0,0 +1,688 @@
+
+"""
+Cool!  Data augmentation is by far the hardest piece of training a classifier
+in the modern era.
+Congratulations on making it this far!
+
+## Optimizer Tuning
+
+To achieve optimal performance, we must implement a Warm up Cosinde decay
+learning rate schedule.
+While we won't go into detail on this schedule, [you can read more about it
+here](https://scorrea92.medium.com/cosine-learning-rate-decay-e8b50aa455b).
+"""
+
+
+def lr_warmup_cosine_decay(
+    global_step,
+    warmup_steps,
+    hold=0,
+    total_steps=0,
+    start_lr=0.0,
+    target_lr=1e-2,
+):
+    # Cosine decay
+    learning_rate = (
+        0.5
+        * target_lr
+        * (
+            1
+            + tf.cos(
+                tf.constant(math.pi)
+                * tf.cast(global_step - warmup_steps - hold, tf.float32)
+                / float(total_steps - warmup_steps - hold)
+            )
+        )
+    )
+
+    warmup_lr = tf.cast(target_lr * (global_step / warmup_steps), tf.float32)
+    target_lr = tf.cast(target_lr, tf.float32)
+
+    if hold > 0:
+        learning_rate = tf.where(
+            global_step > warmup_steps + hold, learning_rate, target_lr
+        )
+
+    learning_rate = tf.where(global_step < warmup_steps, warmup_lr, learning_rate)
+    return learning_rate
+
+
+class WarmUpCosineDecay(schedules.LearningRateSchedule):
+    def __init__(self, warmup_steps, total_steps, hold, start_lr=0.0, target_lr=1e-2):
+        super().__init__()
+        self.start_lr = start_lr
+        self.target_lr = target_lr
+        self.warmup_steps = warmup_steps
+        self.total_steps = total_steps
+        self.hold = hold
+
+    def __call__(self, step):
+        lr = lr_warmup_cosine_decay(
+            global_step=step,
+            total_steps=self.total_steps,
+            warmup_steps=self.warmup_steps,
+            start_lr=self.start_lr,
+            target_lr=self.target_lr,
+            hold=self.hold,
+        )
+
+        return tf.where(step > self.total_steps, 0.0, lr, name="learning_rate")
+
+
+"""
+Next let's construct this optimizer:
+"""
+
+total_steps = (9000 // BATCH_SIZE) * EPOCHS
+warmup_steps = int(0.1 * total_steps)
+hold_steps = int(0.45 * total_steps)
+schedule = WarmUpCosineDecay(
+    start_lr=0.05,
+    target_lr=1e-2,
+    warmup_steps=warmup_steps,
+    total_steps=total_steps,
+    hold=hold_steps,
+)
+optimizer = optimizers.SGD(
+    decay=5e-4,
+    learning_rate=schedule,
+    momentum=0.9,
+)
+"""
+At long last, we can now build our model and call `fit()`!
+"""
+
+backbone = keras_cv.models.EfficientNetV2Backbone.from_preset(
+    "efficientnetv2_b0",
+)
+model = keras.Sequential(
+    [
+        backbone,
+        keras.layers.GlobalMaxPooling2D(),
+        keras.layers.Dropout(rate=0.5),
+        keras.layers.Dense(101, activation="softmax"),
+    ]
+)
+
+"""
+When using `MixUp()` and `CutMix()`, using `label_smoothing` in your loss is
+extremely important.
```

Comment:
A sentence of intuition would be helpful here.

Reply:
:alchemy:

Reply:
It seems intuitive that we want to take our labels less seriously when they are the product of this witchcraft! 

How about:

Since the labels produced by `MixUp()` and `CutMix()` are somewhat artificial, we employ label smoothing to prevent the model from overfitting to artifacts of this augmentation process.

Reply:
Done